boost.sh

```diff
@@ -850,6 +856,38 @@ scrunchAllLibsTogetherInOneLibPerPlatform()
     done
 }
 
+buildUniversalStatic()
+{
+    if [[ -n $UNIVERSAL ]]; then
+        if [[ -n $BUILD_IOS ]]; then
+            echo "Creating universal static library..."
+            mkdir -p "$IOS_BUILD_DIR/universal"
+
+            for NAME in $BOOTSTRAP_LIBS; do
+                if [ "$NAME" == "test" ]; then
+                    NAME="unit_test_framework"
+                fi
+
+                ARCH_FILES=""
+                for ARCH in ${IOS_ARCHS[@]}; do
+                    ARCH_FILES+=" $IOS_BUILD_DIR/$ARCH/libboost_$NAME.a"
+                done
+                # Ideally IOS_ARCHS contains i386 and x86_64 and simulator build steps are not treated out of band
+                if [ -f $IOS_BUILD_DIR/i386/libboost_$NAME.a ]; then
+                    ARCH_FILES+=" $IOS_BUILD_DIR/i386/libboost_$NAME.a"
+                fi
+                if [ -f $IOS_BUILD_DIR/x86_64/libboost_$NAME.a ]; then
+                    ARCH_FILES+=" $IOS_BUILD_DIR/x86_64/libboost_$NAME.a"
+                fi
+                if [[ ${ARCH_FILES[@]} ]]; then
+                    echo "... $NAME"
+                    $IOS_ARM_DEV_CMD lipo -create $ARCH_FILES -o "$IOS_BUILD_DIR/universal/libboost_$NAME.a" || abort "Lipo $1 failed"
+                fi
+            done
+        fi
```

Comment:
You don't need to actually implement this, since it doesn't currently build with Xcode 9 (but you can if you want!), but please add a `# TODO: tvOS` here. If we're providing a FAT binary containing both ARM and Intel arches (device + simulator) for iOS, we should do the same for tvOS